### PR TITLE
Af add posts

### PR DESCRIPTION
**Pull request by AshleyFoster**

### CODE REVIEW

File: app/controllers/api/posts_controller.rb [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
-      format.json
+    @posts = Post.with_attached_image.order("created_at DESC")
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user = current_user
+    @post.image.attach(io: image_io, filename: "image")
+
+    if @post.save
```
Comment (BlakeWilliams) [inline app/controllers/api/posts_controller.rb]: return the post json?

File: app/views/api/posts/_post.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,11 @@
+json.(
+  post,
+  :id,
+  :created_at,
+  :title,
+  :description,
+  :user,
+)
+puts url_for(post.image)
```
Comment (BlakeWilliams) [inline app/views/api/posts/_post.json.jbuilder]: remove

File: app/views/api/posts/_post.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+json.image_url url_for(post.image)
+
```
Comment (BlakeWilliams) [inline app/views/api/posts/_post.json.jbuilder]: extra newline

File: config/environments/development.rb [hunk 4]
```diff
@@ -16,14 +16,17 @@
     config.cache_store = :null_store
   end
 
-  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3003 }
```
Comment (BlakeWilliams) [inline config/environments/development.rb]: change back to 3000?

File: config/environments/development.rb [hunk 5]
```diff
@@ -16,14 +16,17 @@
...
   config.active_support.deprecation = :log
   config.active_record.migration_error = :page_load
   config.assets.debug = true
   config.assets.quiet = true
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 
-  config.web_console.whitelisted_ips = '192.168.1.123'
+  config.default_url_options = config.action_mailer.default_url_options
+
+  Rails.application.routes.default_url_options[:host] = 'http://192.168.2.21:3000/'
```
Comment (BlakeWilliams) [inline config/environments/development.rb]: Remove?

File: config/environments/development.rb [hunk 6]
```diff
@@ -16,14 +16,17 @@
...
   config.action_mailer.perform_caching = false
+  config.active_storage.service = :local
   config.active_support.deprecation = :log
   config.active_record.migration_error = :page_load
   config.assets.debug = true
   config.assets.quiet = true
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 
-  config.web_console.whitelisted_ips = '192.168.1.123'
+  config.default_url_options = config.action_mailer.default_url_options
```
Comment (BlakeWilliams) [inline config/environments/development.rb]: Remove?

File: db/migrate/20180424194716_add_description_to_posts.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class AddDescriptionToPosts < ActiveRecord::Migration[5.1]
+  def change
+    add_column :posts, :description, :text
```
Comment (BlakeWilliams) [inline db/migrate/20180424194716_add_description_to_posts.rb]: `null: false`?

File: db/migrate/20180424194716_add_description_to_posts.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+    add_column :posts, :posted_at, :datetime
```
Comment (BlakeWilliams) [inline db/migrate/20180424194716_add_description_to_posts.rb]: This is useless, use `created_at`

**Review by BlakeWilliams — APPROVED** [review]

File: db/migrate/20180424194716_add_description_to_posts.rb (hunk 8 above)
Comment (AshleyFoster) [reply db/migrate/20180424194716_add_description_to_posts.rb]: That should already be removed in a different db action.

File: app/controllers/api/posts_controller.rb (hunk 1 above)
Comment (AshleyFoster) [reply app/controllers/api/posts_controller.rb]: Does this look okay?

Comment (BlakeWilliams) [reply app/controllers/api/posts_controller.rb]: ohhh, hah. Because jbuilder. I think you want to turn this to `unless @post.save` to remove the else

**Comment by AshleyFoster** [comment]
Okay I think I got all the fixes.

**Pull request merged**